src/python/pants/goal/run_tracker.py

```diff
@@ -367,12 +374,14 @@ def error(msg):
       print(f'WARNING: Failed to upload stats to {stats_url} due to {msg}', file=sys.stderr)
       return False
 
+    if stats_version not in cls.SUPPORTED_STATS_VERSIONS:
```

Comment:
The version is already validated to be one of the `choices` on l91, so I don't believe this is necessary.

Reply:
I think it is since `post_stats()` is a public method that can potentially be called from random places in the code.

Reply:
So, while this is "public" from a python perspective, pants has its own concept of public APIs. If something is not marked `:API: public` as described here: https://www.pantsbuild.org/deprecation_policy.html , then you can assume that it cannot/should-not be used outside of github.com/pantsbuild/pants.